modules/gapi/src/backends/cpu/gparsers.cpp

```diff
@@ -0,0 +1,332 @@
+    {
+        GAPI_Assert(in_ssd_dims.dims() == 4u); // Fixed output layout
+        GAPI_Assert(m_objSize  == 7);          // Fixed SSD object size
+    }
+
+    void adjustBoundingBox(cv::Rect& boundingBox)
+    {
+        auto w = boundingBox.width;
+        auto h = boundingBox.height;
+
+        boundingBox.x -= static_cast<int>(0.067 * w);
+        boundingBox.y -= static_cast<int>(0.028 * h);
+
+        boundingBox.width += static_cast<int>(0.15 * w);
+        boundingBox.height += static_cast<int>(0.13 * h);
+
+        if (boundingBox.width < boundingBox.height)
+        {
+            auto dx = (boundingBox.height - boundingBox.width);
+            boundingBox.x -= dx / 2;
+            boundingBox.width += dx;
+        }
+        else
+        {
+            auto dy = (boundingBox.width - boundingBox.height);
+            boundingBox.y -= dy / 2;
+            boundingBox.height += dy;
+        }
+    }
+
+    std::tuple<cv::Rect, float, float, int> unitedParse(const size_t& step)
+    {
+        const float* it = m_data + step * m_objSize;
+        float image_id   = it[0];
+        int   label      = static_cast<int>(it[1]);
+        float confidence = it[2];
+        float rc_left    = it[3];
+        float rc_top     = it[4];
+        float rc_right   = it[5];
+        float rc_bottom  = it[6];
+
+        cv::Rect rc;  // Map relative coordinates to the original image scale
+        rc.x      = static_cast<int>(rc_left   * m_size.width);
+        rc.y      = static_cast<int>(rc_top    * m_size.height);
+        rc.width  = static_cast<int>(rc_right  * m_size.width)  - rc.x;
+        rc.height = static_cast<int>(rc_bottom * m_size.height) - rc.y;
+        return std::make_tuple(rc, image_id, confidence, label);
+    }
+
+    int getMaxProposals()
+    {
+        return m_maxProp;
+    }
+
+    cv::Rect getSurface()
+    {
+        return m_surface;
+    }
+
+private:
+    const cv::MatSize m_dims;
+    int m_maxProp = 0, m_objSize = 0;
+    const float* m_data = nullptr;
+    const cv::Rect m_surface;
+    const cv::Size m_size;
+};
+} // namespace parsers
+} // namespace gapi
+
+void parseSSDWL(const cv::Mat&  in_ssd_result,
+                const cv::Size& in_size,
+                const float&    confidence_threshold,
+                const int&      filter_label,
+                std::vector<cv::Rect>& out_boxes,
+                std::vector<int>&      out_labels)
+{
+    cv::gapi::parsers::SSDParser parser(in_ssd_result.size, in_size, in_ssd_result.ptr<float>());
+    out_boxes.clear();
+    out_labels.clear();
+    cv::Rect rc;
+    float image_id, confidence;
+    int label;
+    for (int i = 0; i < parser.getMaxProposals(); ++i)
+    {
+        std::tie(rc, image_id, confidence, label) = parser.unitedParse(i);
+
+        if (image_id < 0.f)
+        {
+            break;    // marks end-of-detections
+        }
+
+        if (confidence < confidence_threshold &&
```

Comment:
Should `||` be here? (and the next conditions in brackets then)

Reply:
Fixed